### PR TITLE
add __get_state_size and __get_parameter_size #38

### LINKED ISSUE #38: some exported functions can't be used safely
I believe `__get_parameter` and `__get_state` can't be used safely without a risk of buffer overflow.
Contrary to`__get_exception_msg`, these two don't take a max length parameter, which makes them pretty much as safe as `gets` used to be in C

### PR DESCRIPTION
**Pull request by trinity-1686a**
these allow for proper allocation when an upper bound on the size of the
value is not known

Thank you for contributing to the `fabric-chaincode-wasm` project!
Please  make sure you have done the following before submitting a PR:

* [x] Ran the unit tests: `make build unit-test` (fail at linking, does not seems to be related to this PR)
* [x] Ran the format check: `./scripts/golinter.sh`
* [x] Passed the Git actions integration tests: These will run on every push and PR
* [x] Added the associated issue number in the commit message title

I wonder if not adding new functions, but instead handling null pointer as a special case (don't write to them, but return length as usual) would be a better design
fix #38

### CODE REVIEW

**Comment by kleash** [comment]
Thanks for the quick PR. Let me add few test cases and then we can merge it.

**Review by kleash — APPROVED** [review]

File: sample-wasm-chaincode/chaincode_example02/rust/src/lib.rs [hunk 1]
```diff
@@ -47,41 +49,43 @@ pub extern "C" fn init(args: i64) -> i64 {
...
-    // Assumption here is account name should not be more than 24 character
-    let first_account = [0; 24];
-    let first_account_name_length;
 
-    let second_account = [0; 24];
-    let second_account_name_length;
+    let first_param_len = unsafe { __get_parameter_size(0) }; //Account 1 name
+    let second_param_len = unsafe { __get_parameter_size(1) }; //Account 1 balance
+    let third_param_len = unsafe { __get_parameter_size(2) }; // Account 2 name
+    let fourth_param_len = unsafe { __get_parameter_size(3) }; // Account 2 balance
```
Comment (MHBauer) [inline sample-wasm-chaincode/chaincode_example02/rust/src/lib.rs:56]: This covers testing, right?

**Review by MHBauer — COMMENTED** [review]
Wow, looks clean.

File: sample-wasm-chaincode/chaincode_example02/rust/src/lib.rs (hunk 1 above)
Comment (kleash) [reply sample-wasm-chaincode/chaincode_example02/rust/src/lib.rs:56]: Yeah, this one and line 157 for `__get_state_size` function

**Pull request merged**